A04_attacks-and-mitigations.md

```diff
@@ -1335,21 +1336,21 @@ on the client policy or the grant associated with the refresh token
 Resource servers may make access control decisions based on the identity of a
 resource owner for which an access token was issued, or based on the identity
 of a client in the client credentials grant. If both options are possible,
-depending on the details of the implementation, a client's identity may be
-mistaken for the identity of a resource owner. For example, if a client is able
+depending on the details of the implementation, a resource server may mistake
+a client's identity for the identity of a resource owner. For example, if a client is able
 to choose its own `client_id` during registration with the authorization server,
 a malicious client may set it to a value identifying an end-user (e.g., a `sub`
 value if OpenID Connect is used). If the resource server cannot properly
-distinguish between access tokens issued to clients and access tokens issued to
-end-users, the client may then be able to access resource of the end-user.
+distinguish between access tokens linked to client resources and access tokens linked to
+end-user resources, the client may then be able to access resources belonging to the end-user.
```

Comment:
This is better.  But I'm still slightly puzzled because the end-user being a person isn't a protocol participant.  Therefore, I don't know what is meant by "resources belonging to the end-user", or how they would manifest themselves in the protocol.